src/firmware_new/boards/frankie_v1/io/proto_multicast_communication.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include "pb.h"
+
+typedef struct ProtoMulticastCommunicationProfile ProtoMulticastCommunicationProfile_t;
+
+/**
+ * Event Flags: used to signal tasks to unblock and run specific actions
+ */
+typedef enum
+{
+    PROTO_UPDATED   = 1 << 0,  // protobuf struct has been updated
+    RECEIVED_PROTO  = 1 << 1,  // new protobuf has been received
+    RECEIVE_TIMEOUT = 1 << 2,  // blocking takes longer than NETWORK_TIMEOUT_MS
+} ProtoMulticastCommunicationEventFlags_t;
+
+/***
+ * TASKS:
+ *
+ *                    Sender Task                             Listener Task
+ *         +-------------------------------+         +------------------------------+
+ *         |                               |         |                              |
+ *         |  wait for NETIF_CONFIGURED    |         |  wait for NETIF_CONFIGURED   |
+ *         |                               |         |                              |
+ *         |  join multicast group         |         |  join multicast group        |
+ *         |                               |         |                              |
+ *     +----> wait for PROTO_UPDATED event |     +----> wait for new proto on netif |
+ *     |   |                               |     |   |                              |
+ *     |   |  acquire lock                 |     |   |  acquire lock                |
+ *     |   |                               |     |   |                              |
+ *     |   |  serialize proto to buffer    |     |   |  deserialize buffer to proto |
+ *     |   |                               |     |   |                              |
+ *     |   |  release lock                 |     |   |  release lock                |
+ *     |   |                               |     |   |                              |
+ *     +----+ send buffer to group         |     +----+ notify RECEIVED_PROTO event |
+ *         |                               |         |                              |
+ *         +-------------------------------+         +------------------------------+
+ *
+ * @param arg The pointer to the allocated communication profile to use for the Task
+ * NOTE: These functions must run as a new Task
+ */
+void io_proto_multicast_sender_Task(void* arg);
```

Comment:
Can these task functions be renamed to `io_proto_multicast_sender_task` or was there some special requirement the `Task` had to be with an uppercase `T` for the task to work?

Reply:
nope no reason, do we prefer it lower case?

Reply:
Yea lower case is more correct according to the style guide.

Reply:
fixed